### PR TITLE
rename the variable for resource validation webhook

### PR DESCRIPTION
**Pull request by nimakaviani**
creating the PR to address breaking change in the webhook following this conversation: https://github.com/knative/pkg/pull/636#pullrequestreview-285248410

### CODE REVIEW

**Comment by nimakaviani** [comment]
staged changes in depending repos in the following PRs:

https://github.com/knative/serving/pull/5474
https://github.com/knative/eventing/pull/1853
https://github.com/knative/sample-controller/pull/99

/cc @mattmoor

File: webhook/webhook.go [hunk 1]
```diff
@@ -52,9 +52,9 @@ var (
 
 // ControllerOptions contains the configuration for the webhook
 type ControllerOptions struct {
-	// WebhookName is the name of the webhook we create to handle
+	// ResourceMutationWebhookName is the name of the webhook we create to handle
```
Comment (n3wscott) [inline webhook/webhook.go]: This are normally called "Mutating Admision Webhooks", so `MutatingWebhookName`?

https://kubernetes.io/docs/reference/access-authn-authz/admission-controllers/

Comment (nimakaviani) [reply webhook/webhook.go]: I added `Resource` to qualify it against mutating webhooks that we may add at some other point. `ResourceMutatingWebhookName`?

**Comment by nimakaviani** [comment]
@matzew ah thanks for pointing that out. wasn't aware of that repo. will apply the fix.

**Pull request merged**